tests/modules/mcuboot/direct_xip/src/main.c

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <ztest.h>
+#include <stdio.h>
+
+
+void test_direct_xip(void)
+{
+	/* Checks if the test application boots up in the direct xIP mode. */
+	zassert_true(true, "Needed to pass.");
```

Comment:
Should we have a check here which actually tests if you are in direct XiPlike using an address range from either partition manager or something else and check it against the program counter `MOV PC,R0` ?

Reply:
Actually, in the default test configuration, the primary slot is booted because both images (one in the primary slot and the other in the secondary slot) have the same version and MCUboot boots the image from the primary slot if versions for both slots are equal.

I can configure the test to always boot from the secondary slot in this test. To achieve it, I need to define a new configuration for the child image, intended for the secondary slot, and define the fw version to be greater than 1.1.1+1. In this case, we could add an additional check in the test case to verify that we are executing code in the secondary slot.

Reply:
Maybe the test could read out the MCUBoot header and use that to distinguish if it's booting from primary or secondary? These are just suggestions and not needed for the PR to go in but I think it could be a good idea to have a test which test that it's actually executing in place.

Reply:
Sure, there are probably different ways of identifying a slot that is currently active in the firmware code. More sophisticated test cases would certainly future-proof this feature.

I think that there is value in having a simple test case that just boots up. It can be used as a component of higher level tests (like system/integration tests) that use the test project to execute their testing steps (e.g. in my previous comment https://github.com/nrfconnect/sdk-nrf/pull/7034#discussion_r865644157).

There is also another way to check which partition is currently active by analyzing the output log:
```
*** Booting Zephyr OS build zephyr-v3.0.0-1146-gecb22a1b442c  ***
I: Starting Direct-XIP bootloader
I: Primary   slot: version=1.1.1+1
I: Secondary slot: version=1.2.1+1
I: Image 0 loaded from the secondary slot
I: Bootloader chainload address offset: 0x88000
*** Booting Zephyr OS build zephyr-v3.0.0-1146-gecb22a1b442c  ***
Running test suite test_mcuboot_direct_xip
===================================================================
START - test_direct_xip
 PASS - test_direct_xip in 0.1 seconds
===================================================================
Test suite test_mcuboot_direct_xip succeeded
===================================================================
PROJECT EXECUTION SUCCESSFUL 
```